alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt

```diff
@@ -105,37 +117,78 @@ class TransportGetAlertsAction @Inject constructor(
             .from(tableProp.startIndex)
 
         client.threadPool().threadContext.stashContext().use {
-            resolve(searchSourceBuilder, actionListener, user)
+            scope.launch {
+                try {
+                    val alertIndex = resolveAlertsIndexName(getAlertsRequest)
+                    getAlerts(alertIndex, searchSourceBuilder, actionListener, user)
+                } catch (t: Exception) {
+                    log.error("Failed to get alerts", t)
+                    if (t is AlertingException) {
+                        actionListener.onFailure(t)
+                    } else {
+                        actionListener.onFailure(AlertingException.wrap(t))
+                    }
+                }
+            }
         }
     }
 
-    fun resolve(
+    /** Precedence order for resolving alert index to be queried:
+     1. alertIndex param.
+     2. alert index mentioned in monitor data sources.
+     3. Default alert indices pattern
+     */
+    suspend fun resolveAlertsIndexName(getAlertsRequest: GetAlertsRequest): String {
```

Comment:
Lets add a unit test to verify name resolution

Reply:
this has been verified in the integ test in the following scenarios:

Verify get alerts for monitor WITHOUT custom alerts index.
Verify get alerts for monitor WITH custom alerts index and request has monitor ID.